fe/fe-core/src/main/java/org/apache/doris/transaction/DatabaseTransactionMgr.java

```diff
@@ -941,7 +941,20 @@ public void finishTransaction(long transactionId, Set<Long> errorReplicaIds) thr
                                 }
                             }
 
-                            if (healthReplicaNum < quorumReplicaNum) {
+                            if (transactionState.isPublishTimeout() && healthReplicaNum > 0) {
+                                // We can not do any thing except retrying,
+                                // because publish task is assigned a version,
+                                // and doris does not permit discontinuous
+                                // versions.
+                                //
+                                // If a timeout happens, it means that the rowset
+                                // that are being publised exists on a few replicas we should go
+                                // ahead, otherwise data may be lost and thre
```

Comment:
```suggestion
                                // ahead, otherwise data may be lost and the
```

Reply:
And why data will be lost?

Reply:
e.g. a be crashes and create a new replica on another be.